pkg/stanza/fileconsumer/file.go

```diff
@@ -74,13 +66,23 @@ func (m *Manager) Start(persister operator.Persister) error {
 }
 
 func (m *Manager) closePreviousFiles() {
-	if len(m.knownFiles) > 4*m.movingAverageMatches {
-		m.knownFiles = m.knownFiles[m.movingAverageMatches:]
+	// rotate the filesets at end of every consume() call
+
+	// m.knownFiles[0] wil be removed
+	// m.knownFiles[1] -> m.knownFiles[0]
+	// m.previousPollFiles -> m.knownFiles[1]
+
+	for _, r := range m.previousPollFiles.Reset() {
```

Comment:
Can get rid of `Reset` and switch to using `Pop`?

```suggestion
	for r := m.previousPollFiles.Pop(); r != nil; r = m.previousPollFiles.Pop() {
```